metplotpy/contributed/weather_regime/plot_weather_regime.py

```diff
@@ -0,0 +1,99 @@
+import matplotlib
+matplotlib.use('agg')
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+from pylab import *
+import cmocean
+from cartopy.mpl.gridliner import LONGITUDE_FORMATTER, LATITUDE_FORMATTER
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+from cartopy.util import add_cyclic_point
+import numpy as np
+import sys
+sys.path.insert(0, "/glade/u/home/kalb/UIUC/METcalcpy/")
+import metcalcpy.util.utils as util
+
+def plot_elbow(K,d,mi,line,curve,plot_title,output_plotname):
+
+    plt.plot(K[int(mi)],d[mi]*-1,'*k')
+    plt.plot(K, np.array(curve), 'kx-')
+    plt.plot(K, line, 'r')
+    plt.plot(K, d*-1, 'b')
+    plt.xlabel('k')
+    plt.ylabel('Sum_of_squared_distances')
+    plt.title(plot_title)
+
+    fmt = 'png'
+    full_output_plot = output_plotname + "." + fmt
+    plt.savefig(full_output_plot,format=fmt,dpi=400, bbox_inches='tight')
+
+
+def plot_eof(eof,wrnum,variance_fractions,lons,lats,output_plotname,plevels):
+
+    middleIndex = int((len(lons) - 1)/2)
+    lon0 = lons[middleIndex]
+
+    remp = wrnum%2
+    if remp==0:
+        nrows = wrnum/2
+    else:
+        nrows = (wrnum+1)/2
+
+    #Plot leading 6 modes
+    tran = ccrs.PlateCarree(central_longitude=lon0)
+    proj = tran
+    fig = plt.figure(figsize=(10,10))
+    for i in np.arange(0,wrnum,1):
+        corr = eof[i]
+        ax1 = fig.add_subplot(nrows,2,i+1,projection=proj)
+        contourf(lons,lats,corr,plevels,transform=ccrs.PlateCarree(),cmap = cmocean.cm.balance,extend='both')
+        if (wrnum - i) <= 2:
+            plt.colorbar(orientation ='horizontal',pad=0.01)  #.set_label(label='m',size=15)
+        ax1.add_feature(cfeature.BORDERS)
+        ax1.add_feature(cfeature.STATES)
+        ax1.add_feature(cfeature.COASTLINE)
+        plt.title('EOF '+str(i+1)+' ('+str(round(variance_fractions[i],2))+' %)')
+
+    plt.tight_layout()
+    fmt='png'
+    full_output_plot = output_plotname + "." + fmt
+    plt.savefig(full_output_plot,format=fmt,dpi=400, bbox_inches='tight')
+
+
+def plot_K_means(inputi,wrnum,lon,lat,perc,output_plotname,plevels):
+
```

Comment:
missing python docs with description of input variables.

Reply:
I'm working on the docs right now